external-providers/java-external-provider/pkg/java_external_provider/filter.go

```diff
@@ -91,6 +94,38 @@ func (p *javaServiceClient) filterDefault(symbols []protocol.WorkspaceSymbol) ([
 	return incidents, nil
 }
 
+func (p *javaServiceClient) filterAnnotated(cond *javaCondition, symbols []protocol.WorkspaceSymbol) ([]provider.IncidentContext, error) {
+	incidents := []provider.IncidentContext{}
+	for _, symbol := range symbols {
+		incident, err := p.convertToIncidentContext(symbol)
+		if err != nil {
+			return nil, err
+		}
+		// TODO: problem: for kind == METHOD || kind == CLASS, the containerName returns name of the method or the class, which is fine, but when
+		// kind == FIELD we would probably want to look for the type of the field, not the name. That is not possible atm
+		kind := strings.ToLower(incident.Variables[KIND_EXTRA_KEY].(string))
+		name := incident.Variables[CONTAINER_NAME].(string)
+		pkg := incident.Variables[PACKAGE].(string)
+		pattern := regexp.MustCompile(cond.Referenced.Pattern)
+		if kind == strings.ToLower(cond.Referenced.Location) {
+			// pattern for annotated classes or methods could be a regex
+			if kind == "class" {
+				// for classes, we need the fully qualified name (package + class name)
+				fqn := strings.Join([]string{pkg, name}, ".")
+				if pattern.Match([]byte(fqn)) && kind == strings.ToLower(cond.Referenced.Location) {
+					incidents = append(incidents, incident)
+				}
+			} else if kind == "method" || kind == "field" {
+				if pattern.Match([]byte(name)) && kind == strings.ToLower(cond.Referenced.Location) {
+					incidents = append(incidents, incident)
+				}
```

Comment:
The `kind == strings.ToLower(cond.Referenced.Location) ` condition in both ifs in lines 115 and 119 will be always true since the same condition exists in line 110

Reply:
Good catch!